caput/util/mpitools.py

```diff
@@ -738,32 +827,42 @@ def transpose_blocks(row_array, shape, comm=_comm):
     return recv_buffer.reshape((*shape[:-1], pc))
 
 
-def allocate_hdf5_dataset(fname, dsetname, shape, dtype, comm=_comm):
+# IO Operations
+# =============
+
+
+def allocate_hdf5_dataset(
+    fname: str,
+    dsetname: str,
+    shape: tuple[int, ...],
+    dtype: npt.DTypeLike,
+    comm: MPI.Comm | None = _comm,
+) -> tuple[int, int]:
     """Create a hdf5 dataset and return its offset and size.
 
     The dataset will be created contiguously and immediately allocated,
     however it will not be filled.
 
     Parameters
     ----------
-    fname : string
+    fname : str
         Name of the file to write.
-    dsetname : string
+    dsetname : str
         Name of the dataset to write (must be at root level).
-    shape : tuple
+    shape : tuple[int, ...]
         Shape of the dataset.
-    dtype : numpy datatype
+    dtype : dtype
```

Comment:
Not `npt.DTypeLike`?

Reply:
Within docstrings, intersphinx mapping seems to find `dtype`, `array_like`, and a few other types like that. Although, ~5% of the time this linking seems to fail and I don't know why. `npt.DtypeLike` is only used for type hints. It's annoyingly confusing.